tests/fast-integration/kcp/client.js

```diff
@@ -81,13 +95,127 @@ class KCPWrapper {
+
+            return JSON.parse(res)
+        } catch (error) {
+            throw new Error(`failed during getOrchestrationsOperationStatus`);
+        }
+    }
+
+    async getOrchestrationStatus (orchestrationID) {
+        // debug(`Running getOrchestrationStatus...`)
+        const args = [`orchestrations`, `${orchestrationID}`, `-o`, `json`]
+        try {
+            let res = await this.exec(args);
+            let o = JSON.parse(res)
+
+            debug(`OrchestrationID: ${o.orchestrationID} (${o.type} to version ${o.parameters.kyma.version}), status: ${o.state}`)
+            
+            let operations = await this.getOrchestrationsOperations(o.orchestrationID)
+            // debug(`Got ${operations.length} operations for OrchestrationID ${o.orchestrationID}`)
+
+            let upgradeOperation = {}
+            if (operations.length > 0) {
+                upgradeOperation = await this.getOrchestrationsOperationStatus(orchestrationID, operations[0].operationID)
+                debug(`OrchestrationID: ${orchestrationID}: OperationID: ${operations[0].operationID}: OperationStatus: ${upgradeOperation.state}`)
+            } else {
+                debug (`No operations in OrchestrationID ${o.orchestrationID}`)
+            }
+
+            return res
+        } catch (error) {
+            throw new Error(`failed during getOrchestrationStatus`);
+        }
+    };
+
+    async ensureOrchestrationSucceeded(orchenstrationID) {
+        // Decides whether to go to the next step of while or not based on
+        // the orchestration result (0 = succeeded, 1 = failed, 2 = cancelled, 3 = pending/other)
+        debug(`Waiting for Kyma Upgrade with OrchestrationID ${orchenstrationID} to succeed...`)
+        try {
+            const res = await wait(
+            () => this.getOrchestrationStatus(orchenstrationID),
+            (res) => res && res.state && (res.state === "succeeded" || res.state === "failed"),
+            1000*60*15, // 15 min
+            1000 * 30 // 30 seconds
+            );
+        
+            if(res.state !== "succeeded") {
+                debug("KEB Orchestration Status:", res);
+                throw(`orchestration didn't succeed in 15min: ${JSON.stringify(res)}`);
+            }
+        
+            const descSplit = res.description.split(" ");
+            if (descSplit[1] !== "1") {
+                throw(`orchestration didn't succeed (number of scheduled operations should be equal to 1): ${JSON.stringify(res)}`);
+            }
+        
+            return res;
+        } catch (error) {
+            throw new Error(`failed during ensureOrchestrationSucceeded`);
+        }
+      }
+
     async exec(args) {
         try {
             const defaultArgs = [
                 `--config`, `${this.kcpConfigPath}`,
             ];
-            let output = await execa(`kcp`, args.concat(defaultArgs));
-            debug(output);
+            // debug([`>  kcp`, defaultArgs.concat(args).join(" ")].join(" "))
```

Comment:
Leaving this here for later debugging...

Reply:
If uncommented, this call will print the `kcp` command with **all** of the flags used. That's a sensitive information and should be omitted.

Reply:
I agree with you. Thats why I uncommented it, though I think it can be useful for debugging locally.

However, we still have the problem you mentioned: think of kcp login failing (for whatever reason) it will print out the entire command, including credentials.

I'm not exactly sure how to solve that, but we should consider this (maybe in another PR).